rootex/core/resource_files/basic_material_resource_file.h

```diff
@@ -51,4 +51,8 @@ class BasicMaterialResourceFile : public MaterialResourceFile
 	void reimport() override;
 	bool save() override;
 	void draw() override;
+	std::string getClassName()
```

Comment:
Why is there a need to do this? Just use virtual functions for class specific stuff.

Reply:
+1